pkg/deploy/elbv2/listener_utils.go

```diff
@@ -51,7 +52,15 @@ func buildSDKAction(modelAction elbv2model.Action) (*elbv2sdk.Action, error) {
 		if err != nil {
 			return nil, err
 		}
-		sdkObj.ForwardConfig = forwardConfig
+		if featureGate.Enabled(config.EnforceSingleTargetGroup) {
+			if len(forwardConfig.TargetGroups) == 1 {
+				sdkObj.TargetGroupArn = forwardConfig.TargetGroups[0].TargetGroupArn
+			} else {
+				return nil, errors.New("The controller is configured to specify single Target Group but has more than one.")
```

Comment:
i think the error message better to be "weighted target groups feature is disabled"